component/common/src/main/java/org/wso2/carbon/identity/smsotp/common/SMSOTPServiceImpl.java

```diff
@@ -357,4 +362,29 @@ private int getTenantId() {
 
         return PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
     }
+
+    /**
+     * Get correlation id of current thread
+     *
+     * @return correlation-id
+     */
+    public static String getCorrelationId() {
+        String correlationId;
+        if (isCorrelationIDPresent()) {
+            correlationId = MDC.get(Constants.CORRELATION_ID_MDC).toString();
+        } else {
+            correlationId = UUID.randomUUID().toString();
+
+        }
+        return correlationId;
+    }
+
+    /**
+     * Check whether correlation id present in the log MDC
+     *
+     * @return whether the correlation id is present
+     */
+    public static boolean isCorrelationIDPresent() {
+        return MDC.get(Constants.CORRELATION_ID_MDC) != null;
```

Comment:
Use StringUtils

Reply:
`MDC.get` function returns an object therefore we wont be able to use `stringutils` here.